src/Datadog.Trace/ExtensionMethods/AsyncExtensions.cs

```diff
@@ -0,0 +1,24 @@
+using System;
+using System.Collections.Generic;
+
+namespace Datadog.Trace.ExtensionMethods
+{
+    internal static class AsyncExtensions
+    {
+        internal static T GetOrSet<T>(this AsyncLocalCompat<T> source, Func<T> setter, T unsetValue = default(T))
```

Comment:
Any reason why this is an extension method and not just an instance method on `AsyncLocalCompat()`?

Reply:
Generally you'd ideally keep implementations/interfaces as clean/minimal (i.e. only require implementation of what is needed) and extend from there (which typically allows you to add new/different concrete implementations more easily).  In this case it's not an interface but an actual concrete object, so it'd only really matter if we ever derived it for example.